core/src/metrics/registry.rs

```diff
@@ -42,6 +42,10 @@ impl MetricsRegistry {
         }
     }
 
+    pub fn mock_new() -> Self {
```

Comment:
I think mock() is enough. Also I take it you prolly cant add #[cfg(debug_assertions)], but just in case

Reply:
I renamed it to `mock`, I agree it's better. Unfortunately we can't add #[cfg(test)] without adding feature flags to all crates in the workspace (`cfg(test)` otherwise doesn't propagate to depedendencies).